helm/distributed-idol/resources/dah_preStop.sh

```diff
@@ -1,45 +1,49 @@
+#! /bin/bash
+
 # BEGIN COPYRIGHT NOTICE
-# (c) Copyright 2023 Micro Focus or one of its affiliates.
+# Copyright 2023-2024 Open Text.
+# 
+# The only warranties for products and services of Open Text and its affiliates and licensors
+# ("Open Text") are as may be set forth in the express warranty statements accompanying such
+# products and services. Nothing herein should be construed as constituting an additional warranty.
+# Open Text shall not be liable for technical or editorial errors or omissions contained herein.
+# The information contained herein is subject to change without notice.
+#
 # END COPYRIGHT NOTICE
+
 {{/* 
 preStop hook is used in mirror mode to allow child engines a chance 
 to run their preStop hooks before the DAH disappears (as they are unable
 to tell the difference between scaledown and undeploy).
 In non-mirror mode this is a no-op
 */}}
 {{- if .Values.setupMirrored }}
-function waitForAci() {
-  exit_code=1
-  while [ $exit_code -ne 0 ]; do
-    curl -o - "http://$1:$2/a=getpid" | grep "<autn:pid>"
-    exit_code=$?
-    if [ $exit_code -ne 0 ]; then
-      sleep 1
-    fi
-  done
-}
 
-dahaciport=${IDOL_DAH_SERVICE_PORT_ACI_PORT:-{{ .Values.dah.aciPort | int }}}
+. "$( dirname "${BASH_SOURCE[0]}" )/common_utils.sh"
 
 function waitForLastEngine() {
-    engines=2
-    wait=30
+    local engines=2
+    local wait=30
     for i in $(seq $wait)
     do
-      engines=$(curl -o - "http://localhost:$dahaciport/a=enginemanagement&engineaction=showstatus" | sed "s@<engine @\n<engine @g" | grep -c "<engine ")
+      engines=$(curl -o - ${HTTP_REQ_PARAMS} "${HTTP_SCHEME}://localhost:${IDOL_DAH_ACI_PORT}/a=enginemanagement&engineaction=showstatus" | sed "s@<engine @\n<engine @g" | grep -c "<engine ")
       if [ "$engines" -eq 1 ]; then
         break
       fi
       sleep 1
     done
 }
 
+. "$( dirname "${BASH_SOURCE[0]}" )/common_utils.sh"
```

Comment:
Intentional to add this twice (here and line 22)?

Reply:
[e3d4c5d](https://github.com/opentext-idol/idol-containers-toolkit/pull/92/commits/e3d4c5d46d2f4602f189348a6ea239c85dcd46cf)